### PR TITLE
SIMSBIOHUB-627: API - Telemetry Cronjob

### PR DESCRIPTION
**Pull request by MacDeluca**
## Links to Jira Tickets

- {Include a link to all applicable Jira tickets}

## Description of Changes

- {List all relevant code changes. Include any changes to the business workflow that might not be obvious to the reviewers of this PR.}

## Testing Notes

- {List any relevant testing considerations, necessary pre-reqs, and areas of the app to focus on. Specifically, include anything that will help the reviewers of this PR verify the code is functioning as expected.}

### CODE REVIEW

File: api/src/utils/task-queue.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    };
+
+/**
+ * Process a list of tasks by queueing them and resolving them in parallel concurrently.
+ *
+ * Conceptually equivalent to:
+ * @example
+ *  const taskArrays = [[promiseA, promiseB], [promiseC, promiseD]];
+ *  for (const tasks of taskArrays) {
+ *    const results = Promise.all(task)
```
Comment (NickPhura) [inline api/src/utils/task-queue.ts]: nit: `tasks` 

`Promise.all(tasks)`

File: api/src/utils/task-queue.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *  }
+ *
+ * @template T - The type of the tasks to process via `asyncWorker`
+ * @template J - The type of the resolved value from `asyncWorker`
+ * @param {T[]} tasks - The tasks to process
+ * @param {asyncWorker<unknown, T, J>} asyncWorker - The worker function that processes each task
+ * @param {number} concurrently - The number of tasks to process concurrently
+ * @returns {Promise<QueueResult<T,J>[]>}
+ */
+export const taskQueue = async <T, J>(
```
Comment (NickPhura) [inline api/src/utils/task-queue.ts]: nit: Might be nice to change `T` and `J` to something more descriptive.  Like `TaskType` and `TaskResponseType` or something.

File: api/src/services/telemetry-services/telemetry-vendor-service.ts [hunk 3]
```diff
@@ -15,6 +15,8 @@ import { TelemetryDeploymentService } from './telemetry-deployment-service';
 /**
  * A service class for working with telemetry vendor data.
  *
+ * @see https://api.vectronic-wildlife.com/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#//getPositions_2
```
Comment (NickPhura) [inline api/src/services/telemetry-services/telemetry-vendor-service.ts]: Is there a better place for this link? Seems slightly random. Somewhere in one of the cronjob files would be better I think.

Comment (NickPhura) [reply api/src/services/telemetry-services/telemetry-vendor-service.ts]: Looks like this link is already in cronjob.ts

File: api/src/services/telemetry-services/telemetry-lotek-service.test.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
+import chai, { expect } from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import * as env from '../../utils/env-config';
+import { getMockDBConnection } from '../../__mocks__/db';
+import { TelemetryLotekService } from './telemetry-lotek-service';
+
+chai.use(sinonChai);
+
+describe.only('TelemetryLotekService', () => {
```
Comment (NickPhura) [inline api/src/services/telemetry-services/telemetry-lotek-service.test.ts]: .only

Comment (NickPhura) [reply api/src/services/telemetry-services/telemetry-lotek-service.test.ts]: Getting a few test errors after removing this

File: api/src/services/telemetry-services/telemetry-vectronic-service.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+   * Creates an instance of TelemetryVectronicService.
+   *
+   * @param {IDBConnection} connection
+   */
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this.vectronicClient = axios.create({
+      paramsSerializer: (params) => qs.stringify(params),
+      baseURL: `${getEnvironmentVariable('VECTRONIC_API_HOST')}/v2`
```
Comment (NickPhura) [inline api/src/services/telemetry-services/telemetry-vectronic-service.ts]: Should the `/v2` portion just be included the env var?

File: api/src/services/telemetry-services/telemetry-vendor-service.ts (hunk 3 above)
Comment (MacDeluca) [reply api/src/services/telemetry-services/telemetry-vendor-service.ts]: Dropped. Yea I don't think this was intentional, looks like I copied this to the wrong file

File: api/src/utils/task-queue.ts (hunk 2 above)
Comment (MacDeluca) [reply api/src/utils/task-queue.ts]: Ok I updated them to TaskType + WorkerResultType

File: api/src/services/telemetry-services/telemetry-vectronic-service.ts (hunk 5 above)
Comment (MacDeluca) [reply api/src/services/telemetry-services/telemetry-vectronic-service.ts]: Updated

**Pull request merged**